### PR TITLE
NN-3737:

### PR DESCRIPTION
**Pull request by JamieDraperUK**
Complete a draft adjudication. This will submit the adjudication details to prison api, store a reference to the submission in
the local database. And finally it will delete the previous draft adjudication record.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/config/ApiExceptionHandler.kt [hunk 1]
```diff
@@ -114,20 +112,6 @@ class ApiExceptionHandler {
       )
   }
 
-  @ExceptionHandler(UnAuthorisedToEditIncidentStatementException::class)
-  fun handleUnAuthorisedToEditIncidentStatementException(e: UnAuthorisedToEditIncidentStatementException): ResponseEntity<ErrorResponse?>? {
```
Comment (JamieDraperUK) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/config/ApiExceptionHandler.kt:118]: No longer a requirement.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/controllers/DraftAdjudicationController.kt [hunk 2]
```diff
@@ -136,4 +136,11 @@ class DraftAdjudicationController {
       draftAdjudication
     )
   }
+
+  @PostMapping(value = ["/{id}/complete-draft-adjudication"])
+  @PreAuthorize("hasAuthority('SCOPE_write')")
+  @ResponseStatus(HttpStatus.CREATED)
+  fun completeDraftAdjudication(@PathVariable(name = "id") id: Long) {
+    draftAdjudicationService.completeDraftAdjudication(id)
```
Comment (JamieDraperUK) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/controllers/DraftAdjudicationController.kt:144]: Probably going to return the reported adjudication in later PRs

File: src/main/resources/db/migration/V2__create_submitted_draft_adjudication_table.sql [hunk 3]
```diff
@@ -0,0 +1,12 @@
+drop table if exists submitted_draft_adjudications;
+
+create table submitted_draft_adjudications
```
Comment (JamieDraperUK) [inline src/main/resources/db/migration/V2__create_submitted_draft_adjudication_table.sql:3]: Going to wait until we know how this is going to be accessed until any indexes are added.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt [hunk 4]
```diff
@@ -123,6 +123,29 @@ class DraftAdjudicationIntTest : IntegrationTestBase() {
       .jsonPath("$.draftAdjudication.incidentStatement.modifiedByDateTime").exists()
   }
 
+  @Test
```
Comment (sp-watson) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt:126]: Could do with a negative test - when the save fails in Prison API.

But that's quite a lot of work, so not now.

**Review by sp-watson — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt (hunk 4 above)
Comment (JamieDraperUK) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageadjudicationsapi/integration/DraftAdjudicationIntTest.kt:126]: Agreed. I'll write one now and submit in the next PR

**Pull request merged**